### PR TITLE
fix: assert readonly props

### PR DESCRIPTION
**Pull request by stefl**
## Description

- Addresses multiple instances of Sonar warnings about not setting React component props as readonly
- Adds an eslint rule to asssert that we mark props as readonly
- Prefers type over interface so that we can us the Readonly helper rather than prefix all type definitions with "readonly" repeatedly
- Where types have changed, ensures the types are exported so that they can be used in tests

[Sonar rule](https://sonarcloud.io/organizations/oaknational/rules?open=typescript%3AS6759&rule_key=typescript%3AS6759)

[43 flagged issues](https://sonarcloud.io/project/issues?impactSoftwareQualities=MAINTAINABILITY&rules=typescript:S6759&issueStatuses=OPEN,CONFIRMED&id=oaknational_ai-experiments)

### CODE REVIEW

File: apps/nextjs/src/components/AppComponents/Chat/markdown.tsx [hunk 1]
```diff
@@ -18,15 +18,17 @@ const MemoizedReactMarkdown: FC<Options> = memo(
     prevProps.className === nextProps.className,
 );
 
+export type ReactMarkdownWithStylesProps = Readonly<{
+  markdown: string;
+  lessonPlanSectionDescription?: string;
+  className?: string;
+}>;
```
Comment (codeincontext) [inline apps/nextjs/src/components/AppComponents/Chat/markdown.tsx:25]: It's nice to extract this to a named type outside the function. Make sense

File: apps/nextjs/src/components/AppComponents/QuizDesigner/SuggestedQuestionCard.tsx [hunk 2]
```diff
@@ -8,21 +8,23 @@ import type { PotentialQuestionsType } from "hooks/useSuggestedQuestions";
...
 import { Icon } from "@/components/Icon";
 
+export type SuggestedLessonCardProps = Readonly<{
+  answer: PotentialQuestionsType[0];
+  dispatch: Dispatch<QuizAppAction>;
+  questionsWrapperRef: React.RefObject<HTMLDivElement>;
+  potentialNewQuestions: PotentialQuestionsType;
+  setPotentialNewQuestions: React.Dispatch<PotentialQuestionsType>;
+  questionRefs: React.MutableRefObject<(HTMLDivElement | null)[]>;
+}>;
```
Comment (codeincontext) [inline apps/nextjs/src/components/AppComponents/QuizDesigner/SuggestedQuestionCard.tsx:18]: 😍

**Review by codeincontext — APPROVED** [review]
Nicely done

File: apps/nextjs/src/components/AppComponents/Chat/markdown.tsx (hunk 1 above)
Comment (stefl) [reply apps/nextjs/src/components/AppComponents/Chat/markdown.tsx:25]: If we like this pattern we might want to do a pass on all components to follow the same approach.

Comment (codeincontext) [reply apps/nextjs/src/components/AppComponents/Chat/markdown.tsx:25]: I'm personally a fan of it! As part of my "you should be able to scan what a function does without too much scrolling" guideline

**Pull request merged**